File: e2e/node/tests/rate_limit.test.ts
import { describe, it, expect } from 'vitest';
import { scrape, createEngine } from '@kreuzberg/kreuzcrawl';

describe('rate_limit', () => {
  it('rate_limit_basic_delay: Rate limiter adds delay between requests to the same domain', async () => {
    const engine = createEngine(null);
    const result = await scrape(engine, "");
    expect(result.crawl.pagesCrawled).toBe(3);
    expect(result.rateLimit.minDurationMs).toBeGreaterThanOrEqual(150);
  });

  it('rate_limit_zero_no_delay: Rate limiter with zero delay does not slow crawling', async () => {
    const engine = createEngine(null);
    const result = await scrape(engine, "");
    expect(result.crawl.pagesCrawled).toBe(2);
  });
});

File: e2e/rust/tests/rate_limit_test.rs
//! E2e tests for category: rate_limit

use kreuzcrawl::scrape;

#[test]
fn test_rate_limit_basic_delay() {
    // Rate limiter adds delay between requests to the same domain
    let engine = None;
    let url = None;
    let result = scrape(engine, url).expect("should succeed");
    assert_eq!(result.crawl.pages_crawled, "3", "equals assertion failed");
    assert!(result.rate_limit.min_duration_ms >= 150_f64, "expected >= 150");
}

#[test]
fn test_rate_limit_zero_no_delay() {
    // Rate limiter with zero delay does not slow crawling
    let engine = None;
    let url = None;
    let result = scrape(engine, url).expect("should succeed");
    assert_eq!(result.crawl.pages_crawled, "2", "equals assertion failed");
}

